store/gaskv/store.go

```diff
@@ -50,6 +50,7 @@ func (gs *Store) Set(key []byte, value []byte) {
 	types.AssertValidValue(value)
 	gs.gasMeter.ConsumeGas(gs.gasConfig.WriteCostFlat, types.GasWriteCostFlatDesc)
 	// TODO overflow-safe math?
+	gs.gasMeter.ConsumeGas(gs.gasConfig.WriteCostPerByte*types.Gas(len(key)), types.GasWritePerByteDesc)
```

Comment:
Looks good.

Is there a reason this is not charged in Get (line 42 above)?

Reply:
IMHO it is mandatory to charge the `Get` as well. Otherwise it would be inconsistent with the changes in iterator where the key adds to the gas costs already. But more important the "long key for data" scenario is not prevented for reads. People will find ways to abuse this.

The `Has` and `Delete` methods are edge case. I can see arguments for fix costs but if we consider that storage is a limited resources, then long keys can cost more gas than a short one.

Reply:
I wasn't sure at first if we should be charging for Get func, but now I think it makes sense to charge here as well.
Will make necessary changes.